[feature](merge-cloud) Make MetaService compatible with selectdb cloud

w41ter opened the pull request:
## Proposed changes

Issue Number: close #xxx

An patch for brpc is added, to maintain compatibility between the old and new versions of the service, when modifying the package name of the proto file.

This patch adds a service option 'secondary_package_name', and the bRPC will add the service with the two package names.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

gavinchou commented on cloud/src/meta-service/meta_server.cpp:
```diff
@@ -75,7 +75,11 @@ int MetaServer::start(brpc::Server* server) {
+
+    brpc::ServiceOptions options;
+    options.ownership = brpc::SERVER_OWNS_SERVICE;
+    options.secondary_package_name = "selectdb";  // the package name of selectdb cloud.
```
make it an config option

w41ter committed:
[feature](merge-cloud) Make MetaService compatible with selectdb cloud

An patch for brpc is added, to maintain compatibility between the old and
new versions of the service, when modifying the package name of the proto
file.

This patch adds a service option 'secondary_package_name', and the bRPC
will add the service with the two package names.

w41ter commented:
run buildall

w41ter commented:
run buildall

morningman commented on cloud/src/common/config.h:
```diff
@@ -145,4 +145,7 @@ CONF_Bool(focus_add_kms_data_key, "false");
 CONF_Int32(txn_store_retry_times, "20");
 
+// The secondary package name of the MetaService.
+CONF_String(secondary_package_name, "selectdb");
```
the default should be empty.
And better to explain why need this config and how to use, for other opensource user who want to expand this service

morningman reviewed (approved):
LGTM

morningman reviewed (approved):
LGTM

w41ter commented:
run buildall

dataroaring reviewed (approved):
LGTM
